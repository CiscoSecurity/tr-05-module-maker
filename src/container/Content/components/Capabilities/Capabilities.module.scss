.capabilitiesWrapper {
  display: flex;
  flex-direction: row;
  align-items: center;
}

.inputDescription {
  margin: 4px;
  flex: 1;
}
