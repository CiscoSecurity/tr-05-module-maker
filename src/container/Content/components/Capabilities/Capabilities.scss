@import "src/globals/constants/colors";

.capabilitiesTip {
  padding-left: 4px;
  color: $gray-color;
}
