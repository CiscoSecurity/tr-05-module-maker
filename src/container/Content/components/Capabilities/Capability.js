import React from "react";
import styles from "./Capabilities.module.scss"

export default function Capability ({ api }){
    return <div>
        <div className={styles.capabilitiesWrapper}>
            {api}
            <input
                type="text"
                className={styles.inputDescription}
                placeholder="Enter description"
                autoComplete="off"
            />
        </div>
    </div>;
}