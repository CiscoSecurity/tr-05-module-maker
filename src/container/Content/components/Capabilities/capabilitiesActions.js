import {
    ADD_CAPABILITY,
    DELETE_CAPABILITY,
    UPDATE_CAPABILITY_DESCR
} from "globals/constants/types";

export const addCapability = (api) => ({
        type: ADD_CAPABILITY,
        payload: {id: api}
})

export const updateCapabilityDescription = (api) => ({
        type: UPDATE_CAPABILITY_DESCR,
        payload: {id: api.id, "description": api.description}
})

export const deleteCapability = (api) => ({
    type: DELETE_CAPABILITY,
    payload: api
})
