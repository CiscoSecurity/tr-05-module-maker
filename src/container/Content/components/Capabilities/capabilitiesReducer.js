import { ADD_CAPABILITY, DELETE_CAPABILITY, UPDATE_CAPABILITY_DESCR }
from "globals/constants/types"

const initialState = [];

export const capabilitiesReducer = (state = initialState, action) => {
    switch (action.type) {
        case ADD_CAPABILITY:
            if (state.filter(
                obj => Object.values(obj).includes(action.payload.id)).length === 0
            ) {
                return state.concat([action.payload])
            } else {
                return state
            }
        case DELETE_CAPABILITY:
            return state.filter((api) => api.id !== action.payload)

        case UPDATE_CAPABILITY_DESCR:
            let item = state.filter((api) => api.id === action.payload.id);
            let elementIndex = state.indexOf(item[0]);
            return state.map((item, index) => {
                    if (index !== elementIndex) {
                        return item
                    }
                    return {
                        ...item,
                        ...action.payload
                    }
                }
                )

        default:
            return state
    }
}
