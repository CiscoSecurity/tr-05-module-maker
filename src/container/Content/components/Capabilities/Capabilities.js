import React from "react";
import { connect } from "react-redux";
import Capability from "./Capability";
import * as Constants from "../../../../globals/constants/constants";


const Capabilities = ({syncCapabilities}) => {
            return (
                <div>
                    <p>Capabilities</p>
                    <fieldset>
                        <p>{Constants.CAPABILITIES_TEXT}</p>
                        {syncCapabilities.map(api => {
                            return <Capability api={api} key={api.id}/>
                        })}
                    </fieldset>
                </div>
            )
        }

const mapStateToProps = state => {
    return {
        syncCapabilities: state.json.capabilities
    }
}


export default connect(mapStateToProps, null)(Capabilities);
