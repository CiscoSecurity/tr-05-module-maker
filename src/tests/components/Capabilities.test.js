import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import Capabilities from "../../container/Content/components/Capabilities/Capabilities";
import React from "react";
import configureStore from 'redux-mock-store';
import toJson from "enzyme-to-json";


const mockStore = configureStore([]);


describe('Capabilities', () => {
    it('should render Capabilities with 1 Capability', () => {
        const store = mockStore({
            capabilities: [{id: 'health'}],
        });

        const wrapper = mount(
            <Provider store={store}>
                <Capabilities/>
            </Provider>);
        expect(wrapper.find('Capability')).toHaveLength(1);
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it('should render Capabilities with several inner entities', () => {
        const store = mockStore({
            capabilities: [{id: 'health'}, {id: 'respond'}]
        });

        const wrapper = mount(
            <Provider store={store}>
                <Capabilities/>
            </Provider>);
        expect(wrapper.find('Capability')).toHaveLength(2);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
});
