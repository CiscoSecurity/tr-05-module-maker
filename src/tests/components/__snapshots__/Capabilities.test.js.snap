// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Capabilities should render Capabilities with 1 Capability 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Capabilities)>
    <Capabilities
      dispatch={[Function]}
      syncCapabilities={
        Array [
          Object {
            "id": "health",
          },
        ]
      }
    >
      <div>
        <p>
          Capabilities
        </p>
        <fieldset>
          <p
            className="capabilitiesTip"
          >
            Select at least one of the Supported APIs to write a description
          </p>
          <Connect(Capability)
            api={
              Object {
                "id": "health",
              }
            }
            key="health"
          >
            <Capability
              api={
                Object {
                  "id": "health",
                }
              }
              updateCapabilityDescription={[Function]}
            >
              <div
                className="capabilitiesWrapper"
              >
                <p>
                  health
                  <span
                    className="required-field"
                  >
                    *
                  </span>
                </p>
                <input
                  autoComplete="off"
                  name="health"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="Enter description"
                  required={true}
                  type="text"
                  value=""
                />
              </div>
            </Capability>
          </Connect(Capability)>
        </fieldset>
      </div>
    </Capabilities>
  </Connect(Capabilities)>
</Provider>
`;

exports[`Capabilities should render Capabilities with several inner entities 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Capabilities)>
    <Capabilities
      dispatch={[Function]}
      syncCapabilities={
        Array [
          Object {
            "id": "health",
          },
          Object {
            "id": "respond",
          },
        ]
      }
    >
      <div>
        <p>
          Capabilities
        </p>
        <fieldset>
          <p
            className="capabilitiesTip"
          >
            Select at least one of the Supported APIs to write a description
          </p>
          <Connect(Capability)
            api={
              Object {
                "id": "health",
              }
            }
            key="health"
          >
            <Capability
              api={
                Object {
                  "id": "health",
                }
              }
              updateCapabilityDescription={[Function]}
            >
              <div
                className="capabilitiesWrapper"
              >
                <p>
                  health
                  <span
                    className="required-field"
                  >
                    *
                  </span>
                </p>
                <input
                  autoComplete="off"
                  name="health"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="Enter description"
                  required={true}
                  type="text"
                  value=""
                />
              </div>
            </Capability>
          </Connect(Capability)>
          <Connect(Capability)
            api={
              Object {
                "id": "respond",
              }
            }
            key="respond"
          >
            <Capability
              api={
                Object {
                  "id": "respond",
                }
              }
              updateCapabilityDescription={[Function]}
            >
              <div
                className="capabilitiesWrapper"
              >
                <p>
                  respond
                  <span
                    className="required-field"
                  >
                    *
                  </span>
                </p>
                <input
                  autoComplete="off"
                  name="respond"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="Enter description"
                  required={true}
                  type="text"
                  value=""
                />
              </div>
            </Capability>
          </Connect(Capability)>
        </fieldset>
      </div>
    </Capabilities>
  </Connect(Capabilities)>
</Provider>
`;
